pkg/app/ops/mysqlensurer/indexes.sql

```diff
@@ -1,6 +1,72 @@
--- TODO: Add all required indexes for PipeCD.
--- index on `updated_at` field on `Application` table in ASC direction
-CREATE INDEX application_updated_at_asc ON Application (updated_at ASC);
+--
+-- Application table indexes
+--
 
--- index on `created_at` field on `Application` table in DESC direction
-CREATE INDEX application_created_at_desc ON Application (created_at DESC);
+-- index on `Disabled` and `UpdatedAt` DESC
+CREATE INDEX application_disabled_updated_at_desc ON Application (Disabled, UpdatedAt DESC);
+
+-- index on `EnvId` ASC and `UpdatedAt` DESC
+ALTER TABLE Application ADD COLUMN EnvId VARCHAR(32) GENERATED ALWAYS AS (data->>"$.env_id");
+CREATE INDEX application_env_id_updated_at_desc ON Application (EnvId, UpdatedAt DESC);
+
+-- index on `Name` ASC and `UpdatedAt` DESC
+ALTER TABLE Application ADD COLUMN Name VARCHAR(50) GENERATED ALWAYS AS (data->>"$.name");
+CREATE INDEX application_name_updated_at_desc ON Application (Name, UpdatedAt DESC);
+
+-- index on `Deleted` and `CreatedAt` ASC
+-- TODO: Reconsider make this Deleted column as STORED GENERATED COLUMN
+ALTER TABLE Application ADD COLUMN Deleted BOOL GENERATED ALWAYS AS (data->>"$.deleted");
+CREATE INDEX application_deleted_created_at_asc ON Application (Deleted, CreatedAt);
+
+-- index on `Kind` ASC and `UpdatedAt` DESC
+ALTER TABLE Application ADD COLUMN Kind INT GENERATED ALWAYS AS (data->>"$.kind");
+CREATE INDEX application_kind_updated_at_desc ON Application (Kind, UpdatedAt DESC);
+
+-- index on `SyncState.Status` ASC and `UpdatedAt` DESC
+ALTER TABLE Application ADD COLUMN SyncState INT GENERATED ALWAYS AS (data->>"$.sync_state.status");
```

Comment:
We are passing `SyncStatus.Status` to the filter so I think we have to update this column name to `SyncStatus.Status`.
https://github.com/pipe-cd/pipe/blob/master/pkg/app/api/grpcapi/web_api.go#L580

If the `.` is not allowed by MySQL we have to convert it to something else.

Reply:
Thanks for pointing out this 🙏 In fact `.` is not allowed by MySQL as you mentioned 😅 Let's me make an issue for that and resolve it later 🙏